srv/components/provisioner/salt_master/config.sls

```diff
@@ -20,6 +20,22 @@ salt_master_config_updated:
     - name: /etc/salt/master
     - source: salt://components/provisioner/salt_master/files/master
 
+# Always start glusterfshsaredstorage before salt-master
+Update glusterfssharedstorage.service:
```

Comment:
it might be not the best place for that (in salt's config). Also in case of any changes systemd daemon should be also reloaded.